pallets/settlement/src/lib.rs

```diff
@@ -406,6 +413,18 @@ decl_event!(
         InstructionRescheduled(IdentityId, InstructionId),
         /// An existing venue's signers has been updated (did, venue_id, signers, update_type)
         VenueSignersUpdated(IdentityId, VenueId, Vec<AccountId>, bool),
+        /// A new instruction with memo field has been created
+        /// (did, venue_id, instruction_id, settlement_type, trade_date, value_date, legs, memo)
+        InstructionCreatedWithMemo(
+            IdentityId,
+            VenueId,
+            InstructionId,
+            SettlementType<BlockNumber>,
+            Option<Moment>,
+            Option<Moment>,
+            Vec<Leg>,
+            Option<InstructionMemo>,
+        ),
```

Comment:
Lets just have this event contain the following field:

1. did
2. venue id
3. instruction id
4. memo: InstructionMemo // No `Option` needed here.

Then only emit this event when there is a memo (i.e. memo is some).